src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php

```diff
@@ -37,6 +40,55 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
     /**
      * {@inheritDoc}
      *
+     * In addition to create the object, try to find the parent relation if one exists.
+     */
+    public function prepareObject(TypeInterface $type, $parent = null)
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent) {
+            return $object;
+        }
+
+        /** @var ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+        $parentMappingField = $this->findParentMapping($parent, $metaData);
+        if (false === $parentMappingField) {
+            return $object;
+        }
+
+        $metaData->setFieldValue($object, $parentMappingField, $parent);
+
+        return $object;
+    }
+
+
+    /**
+     * Find the parent object's property which holds the collection entries
+     *
+     * @param object        $parent   The parent entity.
+     * @param ClassMetadata $metaData Metadata of the object that is being created.
+     *
+     * @return string|boolean The name of the parent mapping field or false if none is found.
+     */
+    protected function findParentMapping($parent, ClassMetadata $metaData)
+    {
+        $parentClass = ClassUtils::getClass($parent);
+
+        foreach ($metaData->associationMappings as $mapping) {
+            if ($mapping['targetEntity'] == $parentClass
+                && $metaData->isSingleValuedAssociation($mapping['fieldName'])
```

Comment:
hm. i guess this really depends on the use case. maybe the parent relation is not explicit in some use cases, or the collections are "virtual" (e.g. all documents with a specific tag). maybe we should add a configuration flag that can make the parent relation required (in the type configuration?) and throw an exception in that case, otherwise silently return. do you think that could make sense?